### DISCUSSION TITLE
Radix seem amazing - when is it production ready ?

### DISCUSSION THREAD

**Discussion by orenmizr** [radix-ui/primitives#817] [Ideas] [answered]
one of the best libs out there!

1. i noticed it is missing components that are essential like select. is there an timeline ? 
2. polymorphic slowing down issues - will it be addressed ?

**Comment by colmtuite** [comment, marked as answer]
Thanks!

1. Select is in development. ETA is roughly the end of September.
2. Can you elaborate on this?

**Reply by jjenzz** [reply]
I'm assuming number 2 is the polymorphic perf issues we've been having. I improved this slightly lately but it is still very much in need of a dedicated effort.

**Reply by marbemac** [reply]
➕ re the bonus third question that's hidden in the original title 😄 

> when is it production ready ?

**Reply by colmtuite** [reply]
Each company will need to answer this for themselves. But imo, it is already production-ready. We've been using Radix in production on advanced apps for a very long time. 100+ companies are already using Radix in production.

**Reply by orenmizr** [reply]
i asked because your component's version for example is 0.0.15. 
i assumed you're using semver which states

_> How do I know when to release 1.0.0?
> If your software is being used in production, it should probably already be 1.0.0. If you have a stable API on which users have come to depend, you should be 1.0.0. If you’re worrying a lot about backwards compatibility, you should probably already be 1.0.0._

plus, you're missing some essential components (select is all over any application).

by the way, someone said somewhere vercel uses radix... is that a real thing ?

**Reply by colmtuite** [reply]
Components are versioned independently, so some are further along than others. We're planning to remove the "Alpha" badge very soon.

> select is all over any application

Select is actively in-development right now. ETA is about 2 months.

> someone said somewhere vercel uses radix... is that a real thing ?

Yep, sure it's a real thing. Vercel are using both Radix and Stitches. Lots of other awesome companies are too, like CodeSandbox, Washington Post, Supabase, Calendso, University of Amsterdam etc.

**Reply by marbemac** [reply]
👍 awesome that's all I was looking for - basically sanity checking that the `0.0.x` / `alpha` versions didn't imply that meaningful breaking changes are on the near horizon. Thank you!

**Reply by orenmizr** [reply]
is there an early select component version... i need to build a POC and radix in one of the contenders :)

**Reply by colmtuite** [reply]
No early version just yet.

**Reply by jjenzz** [reply]
@marbemac since we are still in alpha (soon to be beta) I would recommend locking radix primitive versions in your `package.json` (avoid caret/tilde) to prevent any unforeseen changes in your codebase.

**Reply by Francisco-RP** [reply]
(originally put this as an "answer" but it really should be a reply)

> 1. Select is in development. ETA is roughly the end of September.

Now that almost Jan 2022 (Happy new year!), curious to know what the status of the select it. I have a need for it and I've cobbled together something using radix' dropdown but it would be nice to get a select primitive from radix instead.

All I see is this: https://github.com/radix-ui/primitives/issues/384
But no movement on it I think

**Reply by benoitgrelard** [reply]
We're in the later phase of testing, so it should be here soon.

**Reply by Francisco-RP** [reply]
awesome! looking forward to trying it out.  Thank you!
